source/patterns/@aws-solutions-constructs/aws-wafwebacl-apigateway/test/test.wafwebacl-apigateway.test.ts

```diff
@@ -0,0 +1,361 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as cdk from "@aws-cdk/core";
+import { WafwebaclToApiGateway } from "../lib";
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as waf from "@aws-cdk/aws-wafv2";
+import * as defaults from '@aws-solutions-constructs/core';
+import { SynthUtils } from '@aws-cdk/assert';
+import '@aws-cdk/assert/jest';
+
+function deploy(stack: cdk.Stack) {
+  const inProps: lambda.FunctionProps = {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_10_X,
+    handler: 'index.handler'
+  };
+
+  const func = defaults.deployLambdaFunction(stack, inProps);
+
+  const [_api] = defaults.RegionalLambdaRestApi(stack, func);
+
+  return new WafwebaclToApiGateway(stack, 'test-wafwebacl-apigateway', {
+    existingApiGatewayInterface: _api
+  });
+}
+
+function wrapManagedRuleSet(managedGroupName: string, vendorName: string, priority: number) {
+  return {
+    name: `${vendorName}-${managedGroupName}`,
+    priority,
+    overrideAction: { none: {} },
+    statement: {
+      managedRuleGroupStatement: {
+        name: managedGroupName,
+        vendorName,
+      }
+    },
+    visibilityConfig: {
+      cloudWatchMetricsEnabled: true,
+      metricName: managedGroupName,
+      sampledRequestsEnabled: true
+    }
+  } as waf.CfnWebACL.RuleProperty;
+}
+
+// --------------------------------------------------------------
+// Pattern deployment w/ new Lambda function and default properties
+// --------------------------------------------------------------
+test('Pattern deployment w/ new Lambda function and default props', () => {
+  // Initial Setup
+  const stack = new cdk.Stack();
+  deploy(stack);
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
```

Comment:
Remember - we're dropping all snapshots from the jest tests, just doing correctness checks (from our discussion of V1/V2 coexistence last week)